opensrp-opd/src/main/java/org/smartregister/opd/repository/CheckInRepository.java

```diff
@@ -0,0 +1,186 @@
+    };
+
+    public CheckInRepository(Repository repository) {
+        super(repository);
+    }
+
+    public static void createTable(@NonNull SQLiteDatabase database) {
+        database.execSQL(CREATE_TABLE_SQL);
+        database.execSQL(INDEX_BASE_ENTITY_ID);
+        database.execSQL(INDEX_VISIT_ID);
+        database.execSQL(INDEX_EVENT_ID);
+    }
+
+    public ContentValues createValuesFor(@NonNull org.smartregister.opd.pojos.CheckIn checkIn) {
+        ContentValues contentValues = new ContentValues();
+
+        contentValues.put(CheckIn.ID, checkIn.getId());
+        contentValues.put(CheckIn.EVENT_ID, checkIn.getEventId());
+        contentValues.put(CheckIn.VISIT_ID, checkIn.getVisitId());
+        contentValues.put(CheckIn.BASE_ENTITY_ID, checkIn.getBaseEntityId());
+        contentValues.put(CheckIn.PREGNANCY_STATUS, checkIn.getPregnancyStatus());
+        contentValues.put(CheckIn.HAS_HIV_TEST_PREVIOUSLY, checkIn.getHasHivTestPreviously());
+        contentValues.put(CheckIn.HIV_RESULTS_PREVIOUSLY, checkIn.getHivResultsPreviously());
+        contentValues.put(CheckIn.IS_TAKING_ART, checkIn.getIsTakingArt());
+        contentValues.put(CheckIn.CURRENT_HIV_RESULT, checkIn.getCurrentHivResult());
+        contentValues.put(CheckIn.VISIT_TYPE, checkIn.getVisitType());
+        contentValues.put(CheckIn.APPOINTMENT_SCHEDULED_PREVIOUSLY, checkIn.getAppointmentScheduledPreviously());
+        contentValues.put(CheckIn.APPOINTMENT_DUE_DATE, checkIn.getAppointmentDueDate());
+
+        return contentValues;
+    }
+
+    @Nullable
+    @Override
+    public org.smartregister.opd.pojos.CheckIn getLatestCheckIn(@NonNull String clientBaseEntityId) {
+        Cursor mCursor = null;
+        org.smartregister.opd.pojos.CheckIn checkIn = null;
+        try {
+            SQLiteDatabase db = getWritableDatabase();
+
+            if (StringUtils.isNotBlank(clientBaseEntityId)) {
+                mCursor = db.query(OpdDbConstants.Table.CHECK_IN, columns, CheckIn.BASE_ENTITY_ID + " = ?"
+                        , new String[]{clientBaseEntityId}
+                        , null
+                        , null
+                        , CheckIn.CREATED_AT + " DESC"
+                        , "1");
+
+                if (mCursor != null) {
+                    if (mCursor.moveToNext()) {
+                        checkIn = getCheckInResult(mCursor);
+                    }
+                }
+            }
+
+        } catch (Exception e) {
+            Timber.e(e);
+        } finally {
+            if (mCursor != null) {
+                mCursor.close();
+            }
+        }
+
+        return checkIn;
+    }
+
+    @Nullable
+    @Override
+    public org.smartregister.opd.pojos.CheckIn getCheckInByVisit(@NonNull int visitId) {
+
+        Cursor mCursor = null;
+        org.smartregister.opd.pojos.CheckIn checkIn = null;
+        try {
+            SQLiteDatabase db = getWritableDatabase();
+
+            if (visitId != 0) {
```

Comment:
Why are you checking for 0 here?

Reply:
default int value, since it can't be null. I have removed the @NonNull notation which will not work in the case of primitive types

Reply:
Cool